### ISSUE TITLE
Callback IDs in ActionScript are stored on the instance and overwritten.

### ISSUE THREAD

**Issue by gtanner** [blackberry/WebWorks-TabletOS#5]
Example Case:

```
//I am expecting this to be successful
blackberry.payment.purchase({digitalGoodName: "Successful Purchase"}, function () {
     console.log("win 1");
},
function () {
     console.log("fail 1");
});

blackberry.payment.purchase({digitalGoodName: "Unsuccessful Purchase"}, function () {
     console.log("win 2");
},
function () {
     console.log("fail 2");
});
```

What will Happen:

```
Console:
win 2
fail 2
```

That is not good!  The reason for this is the Actionscript code stores the callback IDs in instance variables and trumps them each call.

See:
https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.payment/src/Air/Payment/src/blackberry/payment/Payment.as#L33-34

Which is trumped each call here:
https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.payment/src/Air/Payment/src/blackberry/payment/Payment.as#L83-84

This problem can be observed in the following APIs:
## Dialog

https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.ui.dialog/src/Air/UserInterface/src/blackberry/ui/dialog/Dialog.as#L53
## Camera

https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.media.camera/src/Air/Camera/src/blackberry/media/camera/Camera.as#L37-39
## Microphone

https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.media.microphone/src/Air/Microphone/src/blackberry/media/microphone/MicrophoneExtension.as#L39-40
## Payment

https://github.com/blackberry/WebWorks-TabletOS/blob/master/js_api/blackberry.payment/src/Air/Payment/src/blackberry/payment/Payment.as#L33-34

---

As a general rule I would expect all of the actionscript code to be stateless and not store anything in instance variables as we need to think of the actionscript as a stateless webserver.

**Comment by gtanner** [comment]
We ran into this issue while developing the barcode extension and fixed it by closuring the callback IDs
see: https://github.com/tneil/MessingAround/commit/ce2c4054580fd43db4c9ab71903e701042d50c6c

**Comment by cdelcol** [comment]
Fixed with Pull request #34.

Closed.
